tests/rpc/test_rpc_emc.py

```diff
@@ -0,0 +1,453 @@
+"""
+Unit test file for rpc/external_message_consumer.py
+"""
+import asyncio
+import functools
+import logging
+from datetime import datetime, timezone
+from unittest.mock import MagicMock
+
+import pytest
+import websockets
+
+from freqtrade.data.dataprovider import DataProvider
+from freqtrade.rpc.external_message_consumer import ExternalMessageConsumer
+from tests.conftest import log_has, log_has_re, log_has_when
+
+
+_TEST_WS_TOKEN = "secret_Ws_t0ken"
+_TEST_WS_HOST = "localhost"
+_TEST_WS_PORT = 9989
+
+
+@pytest.fixture
+def patched_emc(default_conf, mocker):
+    default_conf.update({
+        "external_message_consumer": {
+            "enabled": True,
+            "producers": [
+                {
+                    "name": "default",
+                    "url": "ws://127.0.0.1:8080/api/v1/message/ws",
+                    "ws_token": _TEST_WS_TOKEN
+                }
+            ]
+        }
+    })
+    dataprovider = DataProvider(default_conf, None, None, None)
```

Comment:
I suspect that all tests that use this fixture can be flaky.

websocket communication is not mocked - so they'll depend on nothing running on port 8080, or responding in ways the tests don't expect.
you can easily test this by running freqtrade on 8080 while running tests in this file - the log will contain connection requests.

Reply:
The actual connections testing aren't using that fixture. I'd rather not have to run a whole instance just to run this test file.

Reply:
my point is exactly the reverse though - having an instance running makes pytest connect to it (visible in it's logs) - which is wrong as well.
i suspect it's this fixture (maybe it's something else in this file) - but it shouldn't try to connect to 127.0.0.1:8080 at all from tests - as that makes tests flaky (we can't know what people have on 8080 on their machines - and randomly connecting to that would be wrong.

Reply:
Oh I see alright, so I just need to change those lines to something completely random